content/root/best-practices-guide.textile

```diff
@@ -0,0 +1,97 @@
+
+* Ably pushes all the data published on a channel to every subscriber attached to that channel. Subscribers can choose to only subscribe to events with a certain message name, but that filtering is done on the client side (however, it's useful for things like letting the subscriber trigger different listeners for different events). If you have messages that don't need to be sent to some clients at all, you're often better off using a separate channel for those instead of implementing a filter. This way, you won't be sending unnecessary messages that count towards your package limit.
+
+h4. 3. Being present without subscribing to presence events
+
+* If you want lots of people to be present, but not necessarily listen for "presence change events":/realtime/presence#presence-states, you should use token capabilities to prevent all users from receiving presence notifications while allowing them to enter only. This is a common scenario when you just want to know how many people are present on a channel at any point in time.
+
+h4. 4. Handling connection and channel state&#45;change events
+
+* Ably's Realtime SDKs automatically handle disruptions such as network disconnections, so you don't need to do anything. The library emits "connection":/realtime/connection#connection-states and "channel":/realtime/channels-messages#channel-stateschannel state change events that you can listen to in case you'd like be informed about what it's doing (for example, if you want to know whenever the library has lost message continuity on a channel due to being disconnected for more than two minutes). But there's no need to take any particular action in response to any of these, the library will automatically reconnect and reattach to channels after the network disruption is over.
+
+h4. 5. Using API key on the client side
+
+* Ably's "Basic Authentication":/general/authentication#basic-authentication method is the easiest way to authenticate your client on the Ably Realtime platform. However, it directly exposes the API key in the source code. When you use it on the client side, any entity that interacts with the client could view your API key and potentially misuse it. If you can run an auth server, we recommend using "Token Authentication":/general/authentication#token-authentication on the client side. "Read more":/general/authentication#selecting-auth about selecting an authentication mechanism.
+
+h4. 6. Ensuring tokens are auto&#45;renewable
+
+* Tokens are short lived and need to be constantly renewed. Rather than getting a token from your auth servers yourself and initializing the Client Library SDK with it, we recommend you instead initialize the SDK with an authUrl or authCallback. It will use this information to automatically request a token from your auth server's endpoint whenever it needs a new one, thus seamlessly switching to the new token without disconnecting, hence ensuring an uninterrupted service.
```

Comment:
I don't think this description is right.
We should really be saying "It is almost never correct to provide a token to a client library without ensuring the client library has a means to automatically get a new token automatically when it needs one (tokens expire as they are intentionally short lived). The Ably client library SDKs provide two mechanisms in the `ClientOptions` (linked) so that it can obtain tokens automatically, via `authUrl` and `authCallback`.  If you are using token authentication, ensure your client has a means to renew tokens by using `authUrl` or `authCallback`."  Perhaps then provide a simple example and links to appropriate documentation.